src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1181,13 +1180,16 @@ CheckCitusVersion(int elevel)
  * this function logs an error with the specified elevel and returns false,
  * otherwise it returns true.
  */
-static bool
+bool
 CheckAvailableVersion(int elevel)
 {
 	char *availableVersion = NULL;
 
-	Assert(CitusHasBeenLoaded());
-	Assert(EnableVersionChecks);
+	if (citusVersionKnownCompatible ||
```

Comment:
I think this will only help to eliminate for the first non-cached version check after CREATE/ALTER EXTENSION. However, it is nice that we can make citusVersionKnownCompatible private. I changed it.